### PR TITLE
[DOC] Write local documentation developer guide.

### PR DESCRIPTION
**Pull request by aadya940**
Write steps to build documentation locally.

### CODE REVIEW

**Comment by MatthewMiddlehurst** [comment]
Mind if i make changes to this? Quite a bit that can be improved on this page IMO.

**Comment by aadya940** [comment]
@MatthewMiddlehurst Yes sure, were you able to build the docs?

File: docs/developer_guide/documentation.md [hunk 1]
```diff
@@ -125,7 +125,26 @@ pip install .[docs]
 
 ```{code-block} powershell
 cd docs
+sphinx-apidoc -o docs/ ../
+```
+
+NOTES:
+- This will auto-generate documentation by recursively travelling through the modules of `aeon` as static `.rst` files by parsing the docstrings. Hence, make sure your contribution is accessible through `__all__` if it exists.
+- Whenever you write new documentation, for example a class, mention it in the `docs/api_reference`.
+- Whenever you write new documentation, rerun `sphinx-apidoc` command after deleting the old generated files.
```
Comment (MatthewMiddlehurst) [inline docs/developer_guide/documentation.md:134]: I also dont think this is needed, refreshing after a new `make.bat html` worked for me.

**Comment by aadya940** [comment]
@MatthewMiddlehurst  Oh, Okay! I build my docs in the way I wrote and have written this as per my experience. However, if this works, it would be much simpler. :))

**Comment by aadya940** [comment]
Could you please edit accordingly?

**Comment by MatthewMiddlehurst** [comment]
absorbed into #2297. Includes your commit so you should still get credit

**Comment by MatthewMiddlehurst** [comment]
@all-contributors add @aadya940 for docs

**Pull request closed without merging**